contracts/validators/CredentialAtomicQueryValidator.sol

```diff
@@ -6,79 +6,100 @@ import {GenesisUtils} from "../lib/GenesisUtils.sol";
 import {ICircuitValidator} from "../interfaces/ICircuitValidator.sol";
 import {IVerifier} from "../interfaces/IVerifier.sol";
 import {IState} from "../interfaces/IState.sol";
+import {PoseidonFacade} from "../lib/Poseidon.sol";
 
 abstract contract CredentialAtomicQueryValidator is OwnableUpgradeable, ICircuitValidator {
-    IVerifier public verifier;
+    struct CredentialAtomicQuery {
+        uint256 schema;
+        uint256 claimPathKey;
+        uint256 operator;
+        uint256 slotIndex;
+        uint256[] value;
+        uint256 queryHash;
+        uint256[] allowedIssuers;
+        string[] circuitIds;
+        bool skipClaimRevocationCheck;
+    }
+
+    // This empty reserved space is put in place to allow future versions
+    // of the CredentialAtomicQuerySigValidator contract to inherit from other contracts without a risk of
+    // breaking the storage layout. This is necessary because the parent contracts in the
+    // future may introduce some storage variables, which are placed before the CredentialAtomicQuerySigValidator
+    // contract's storage variables.
+    // (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps)
+    // slither-disable-next-line shadowing-state
+    // slither-disable-next-line unused-state
+    uint256[500] private __gap_before_base;
+
+    mapping(string => IVerifier) internal _circuitIdToVerifier;
+    string[] internal _supportedCircuitIds;
+
     IState public state;
 
-    uint256 public revocationStateExpirationTime;
-    uint256 public proofGenerationExpirationTime;
+    uint256 public revocationStateExpirationTimeout;
+    uint256 public proofExpirationTimeout;
+    uint256 public gistRootExpirationTimeout;
+
+    mapping(string => uint256) internal _inputNameToIndex;
+
+    // This empty reserved space is put in place to allow future versions
+    // of this contract to add new variables without shifting down
+    // storage of child contracts that use this contract as a base
+    // (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps)
+    uint256[44] __gap_after_base;
```

Comment:
This be better of the type `uint256[43]`. Check the layout to be a multiple of 50